src/lib.rs

```diff
@@ -76,20 +60,54 @@ pub fn calculate_witness(
         }
     }
 
-    // Prepare inputs
     let mut inputs = vec![U256::ZERO; max_index + 1];
     inputs[0] = U256::from(1);
+    inputs
+}
 
-    // Set input values from JSON
-    for (key, value) in input_list {
+/// Calculates the position of the given signal in the inputs buffer
+pub fn get_input_mapping(
+    input_list: HashMap<String, Vec<U256>>,
```

Comment:
Given only the keys are used here might be better off passing a `input_labels: &Vec<String>`

Reply:
Here's what we'd do with the current API (not ideal):

```
    let variable_names = [
      "prog_a_rw", 
      "prog_v_rw", 
      "memreg_a_rw", 
      "memreg_v_reads", 
      "memreg_v_writes", 
      "chunks_x", 
      "chunks_y", 
      "chunks_query", 
      "lookup_output", 
      "op_flags"
    ];

    let graph = witness::init_graph(WTNS_GRAPH_BYTES).unwrap();
    let mut variable_inputs: HashMap<String, Vec<U256>> = HashMap::new();
    for name in &variable_names {
        variable_inputs.insert(name.to_string(), vec![]);
    }
    let input_mapping = witness::get_input_mapping(variable_inputs, &graph);
```

Swapping to &Vec<String> would skip that empty vec work.